caikit/core/registries.py

```diff
@@ -0,0 +1,75 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Single spot where stick accessors for global shared state
+
+(Shining the light on the shared state so that we can clean it up and re-home it)
+This SHOULD NOT import anything else from caikit outside the toolkit error handler
+"""
+
+# Standard
+from typing import Any, Dict, Tuple, Type
+
+# First Party
+import alog
+
+# Local
+from caikit.core.toolkit.errors import error_handler
+
+log = alog.use_channel("REGISTRIES")
+error = error_handler.get(log)
+
+
+# Registry of all classes decorated with `@caikit.module`
+MODULE_REGISTRY = {}
+
+# Global module backend registry dict
+MODULE_BACKEND_REGISTRY = {}
+
+
+def module_registry() -> Dict[str, "caikit.core.ModuleBase"]:
```

Comment:
Wondering if we want to move to separating write access somehow? Probably not for this PR

Reply:
Would be nice to get docstrings on these with appropriate warnings about mutating global state!

Reply:
true!

Reply:
done, will leave to followup PR to clean these up with some safer setters